File: R/dir_SWC2ABA_plots/select_file_csv.R
clr <- function() cat("\014")

select_file_csv <- function() {
  fileDir <- "data"
  
  csvFiles <- list.files(fileDir, pattern="*.csv")
  nCsvFiles <- length(csvFiles)
  print(nCsvFiles)
  
  reply <- ""
  
  # main loop to display menu choices and accept input
  # terminates when user chooses to exit
  while (reply == "") {
    ## display menu ##
    
    clr() 
    
    strng <- "Please select your .csv file from the selections below:\n"
    print(strng)
    cat("\n")
    
    for (i in 1:nCsvFiles) {
      strng <- sprintf("    %2d) %s", i, csvFiles[[i]])
      print(strng)
    }   
    
    print("     !) Exit")
    cat("\n")
    
    reply <- readline(prompt="Your selection: ")
    
    ## process input ##
    
    if (reply == "!") {
      csvFileName <- "!"
    } else {
      num <- as.numeric(reply)
      
      if (num <= nCsvFiles) {
        csvFileName <- sprintf("./data/%s", csvFiles[[num]])
      } else {
        reply <- ""
      } # if num
    } # if reply
    
  } # while loop
  
  return(csvFileName)
  
} # select_file_csv

File: MATLAB/dir_SWC2ABA_plots/lib/select_file_csv.m
function csvFileName = select_file_csv()

    fileDir = sprintf('./data/*.csv');
    
    csvFiles = dir(fileDir);
    nCsvFiles = length(csvFiles);

    reply = [];
    
    % main loop to display menu choices and accept input
    % terminates when user chooses to exit
    while (isempty(reply))
        %% display menu %%
        
        clc;
        
        strng = sprintf('Please select your .csv file from the selections below:\n');
        disp(strng);

        for i = 1:nCsvFiles
            strng = sprintf('    %2d) %s', i, csvFiles(i).name);
            disp(strng);
        end % for i
        
        disp('     !) Exit');
        
        reply = input('\nYour selection: ', 's');

        
        %% process input %%
        
        if strcmp(reply, '!')
            csvFileName = '!';
        else 
            num = str2double(reply);
            
            if num <= nCsvFiles
                csvFileName = sprintf('./data/%s', csvFiles(num).name);
            else
                reply = [];
            end % if num
        end
        
    end % while loop

end % select_file_csv()